README.md

```diff
@@ -65,67 +105,173 @@ assert new_obj == obj
+
+@register_extensions('bar')
+class BarParser(Parser):
+    @classmethod
+    def load_content(cls, content: bytes):
+        # Parse bytes from "bar" file format into a Python object
+        ...
+
+    @classmethod
+    def dump_content(cls, data: Any):
+        # Dump a Python object into bytes in the "bar" file format
+        ...
 ```
 
-## Protocol Configuration
+Now if we redo our above example using the `.bar` extension:
+
+```python
+from ... import FooCabinet, BarParser  # ensure FooCabinet and BarParser are loaded
 
-Some storage platform protocols may require some configuration parameters to be set before they can be used.
-Each `Cabinet` subclass can expose a `set_configuration(**config)` classmethod to take 
-care of any required initial setup.
+cabinets.read('foo://example.bar')
+```
+
+This statement is roughly equivalent to:
 
 ```python
-from cabinets.cabinet.s3 import S3Cabinet
+BarParser.load_content(FooCabinet._read_content('foo.bar'))
+```
 
-# set the AWS S3 region to us-west-2 and specify an access key
-S3Cabinet.set_configuration(region_name='us-west-2', aws_access_key_id=...)
+and should return a Python object from your `Foo` cabinet, using your `Bar` parser!
+
+### Loading Plugins
+
+As mentioned in the example above, your custom `Cabinet` and `Parser` classes must be
+executed in order to be added to the internal cache `cabinets` uses for extension and
+protocol lookup. If your custom classes are imported before any `cabinets` functions are
+used then this won't be an issue. However, in many use cases there is no reason to
+import those classes aside from usage with `cabinets` functions. Instead of requiring
+each class to be imported manually at the start of your program,
+`cabinets` can search a specified path for new `Cabinet` and `Parser` classes, and load
+them automatically.
+
+If the environment variable `PLUGIN_PATH` is specified, `cabinets`
+will search that path for subdirectories called `cabinet` and `parser`. Modules residing
+within those directories will be searched for `Cabinet` and `Parser` subclasses,
+respectively.
 
-# use specific Cabinet to avoid protocol prefix
-S3Cabinet.read('bucket-in-us-west-2/test.json') 
-# or use generic Cabinet with protocol prefix
-import cabinets
-cabinets.read('s3://bucket-us-west-2/test.json')
 ```
+└─ PLUGIN_PATH
+    |
+    └───cabinet
+    │   │   custom_cabinet.py   
+    └───parser
+    │   │   custom_parser_1.py
+    │   │   custom_parser_2.py
+```
+
+If the above `FooCabinet` and `BarParser` classes are placed in
+`custom_cabinet.py` and `custom_parser_1.py` (or `custom_parser_2.py`) they will be
+loaded and registered to their specified protocols/extensions without needing to be
+referenced anywhere else in the program.
+
+## Contributing
+
+This package is open source (see [LICENSE](LICENSE)), so please feel free
+to [contribute](https://opensource.guide/how-to-contribute/)
+by submitting a pull request, creating an issue, or contacting the authors directly.
```

Comment:
Are there settings we need to enable to allow people to push branches and submit PRs? Remember you couldn't push until I made you a contributor.